TestServer/server.mjs

```diff
@@ -52,6 +51,43 @@ app.use('/load_latency/assets/', (req, __, next) => {
     }, delay);
 });
 
+//
+// XHR latency simulation for networkidle tests
+//
+app.get('/xhr_latency/:ms/*.html', (req, res) => {
+    const delay = parseInt(req.params.ms, 10);
+    if (isNaN(delay)) {
+        res.status(400).send('Invalid latency parameter');
+        return;
+    }
+
+    const fileName = req.params[0];
+    const fullPath = path.resolve(path.join('public', 'xhr_latency', fileName + '.html'));
+    const fileContents = fs.readFileSync(fullPath, 'utf8');
+    const modifiedContents = fileContents.replace('{{delay}}', delay.toString());
+    res.contentType('text/html').send(modifiedContents);
+});
+
+// Middleware to apply latency to specific XHR requests
+app.use('/xhr/', (req, __, next) => {
+    // parse ?delay= from req
+    const delay = parseInt(req.query.delay, 10);
+    if (isNaN(delay)) {
+        next();
+        return;
+    }
+
+    setTimeout(() => {
+        next();
+    }, delay);
+});
+
+//
```

Comment:
cleanup

Reply:
This was semi-intentional (it's just in the wrong place :)), but how would you go about separating different sections of the file without splitting it into separate files (e.g. here I wanted to have the group of load latency be visually separate from xhr latency group). I've seen folks do `////////////////////////////////////////////` but I don't like that. So I through just more whitespace between the sections.

I'm totally fine with not visually separating anything, it was more for aesthetics.